### PR TITLE
[GH-113] Unit tests with TS for various files

### PR DESCRIPTION
**Pull request by shubham43MP**

### CODE REVIEW

File: templates/base/src/__tests__/_404.test.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import Custom404 from '../pages/404';
+
+test('404 Error Page ', () => {
+  let { container, getByText, rerender } = render(<Custom404 />);
+  expect(getByText('404 - Page Not Found')).toBeInTheDocument();
+  expect(container.firstChild).toMatchInlineSnapshot(`
```
Comment (vinodloha) [inline templates/base/src/__tests__/_404.test.tsx:9]: Delete Snapshot.

File: templates/base/src/__tests__/_404.test.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import Custom404 from '../pages/404';
+
+test('404 Error Page ', () => {
+  let { container, getByText, rerender } = render(<Custom404 />);
+  expect(getByText('404 - Page Not Found')).toBeInTheDocument();
```
Comment (vinodloha) [inline templates/base/src/__tests__/_404.test.tsx:8]: Good to have] getByText. --> getByHeading

File: templates/base/src/__tests__/_app.test.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+describe('Root App rendering check', () => {
+  test('should render correctly', () => {
+    expect(true).toBe(true);
```
Comment (vinodloha) [inline templates/base/src/__tests__/_app.test.ts:6]: This should be changed to a more real test, like existence of component.

File: templates/base/src/__tests__/_document.test.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { render } from '@testing-library/react';
+import MyDocument from '../pages/_document';
+
+describe('<MyDocument />', () => {
+  it('should render without errors', async () => {
+    expect(true).toBe(true);
```
Comment (vinodloha) [inline templates/base/src/__tests__/_document.test.ts:6]: Check for component existence here.

File: templates/common/src/components/atoms/Link/tests/Link.test.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+
+it('Testing the Link Router', () => {
+  expect(true).toBe(true);
```
Comment (vinodloha) [inline templates/common/src/components/atoms/Link/tests/Link.test.ts:6]: Fix this

File: templates/base/src/__tests__/_404.test.tsx (hunk 2 above)
Comment (shubham43MP) [reply templates/base/src/__tests__/_404.test.tsx:8]: In the API, we have getByTitle not exactly something that fetches heading element directly

File: templates/base/src/__tests__/_document.test.ts (hunk 4 above)
Comment (shubham43MP) [reply templates/base/src/__tests__/_document.test.ts:6]: @vinodloha  Not Able to render it like a normal react based component with Testing library's render method. It shows error that a class component cant be called without new

**Review by vinodloha — APPROVED** [review]

**Pull request merged**